### ISSUE TITLE
Standalone - Total Nuclear Annihilation - plastics / music issues

### ISSUE THREAD

**Issue by francisdb** [jsm174/vpx-standalone-scripts#22]
Something is messed up with the fluorescent plastic rendering and with the dmd ()

This is what I see

![image](https://github.com/jsm174/vpx-standalone-scripts/assets/161305/76fb5f91-3382-4b31-aec5-88b484516ae4)

<img width="268" alt="image" src="https://github.com/jsm174/vpx-standalone-scripts/assets/161305/ea425a25-cde2-4d09-9245-5d845c7951a4">

This is what you see with standard vpx

<img width="763" alt="image" src="https://github.com/jsm174/vpx-standalone-scripts/assets/161305/12dd8c1b-6b25-4c8b-8595-28b61de832f9">

**Comment by francisdb** [comment]
One more issue, the soundtrack I bought is not playing.
You mentioned you do path fixing in the music player. However the script does a `fso.FolderExists` which returns false.
So to get music on mac you need this patch

```diff
--- /Users/francisdb/vpinball/tables/Total Nuclear Annihilation (Spooky 2017) VPW v2.2/Total Nuclear Annihilation (Spooky 2017) VPW v2.2.vbs.original.tmp	2023-06-12 15:27:15
+++ /Users/francisdb/vpinball/tables/Total Nuclear Annihilation (Spooky 2017) VPW v2.2/Total Nuclear Annihilation (Spooky 2017) VPW v2.2.vbs	2023-06-12 15:27:12
@@ -64,8 +64,8 @@
 ' \Visual Pinball\Music\TNA
 ' then you must set bMusicOn to True in the script:

-Const 	Disable_TNA_Message		= False		'True - to disable the TNA pop-up message.  False = display TNA pop-up message
-Const 	bMusicOn				= False  	'True = enable, False = disable TNA original music
+Const 	Disable_TNA_Message		= True		'True - to disable the TNA pop-up message.  False = display TNA pop-up message
+Const 	bMusicOn				= True  	'True = enable, False = disable TNA original music
 Const 	bMusicVolume			= 0.7		'A volume multiplier for the game's music (Applicable only if bMusicOn=true)

 ' Please show Scott Danesi your support and thanks for creating an awesome pinball game and for making the music available for us to buy.
@@ -1484,9 +1484,9 @@
 sub InitMusicFolder
 	Dim fso
 	Set fso = CreateObject("Scripting.FileSystemObject")
-	if Not fso.FolderExists(MusicDirectory & "\" & MusicSubDirectory) Then
-		MusicSubDirectory = ""
-	end if
+	' if Not fso.FolderExists(MusicDirectory & "\" & MusicSubDirectory) Then
+	' 	MusicSubDirectory = ""
+	' end if
 	set fso = nothing
 end sub


```

**Comment by francisdb** [comment]
Forget about the DMD part, only the plastics issue remains

**Comment by francisdb** [comment]
![image](https://github.com/jsm174/vpx-standalone-scripts/assets/161305/87a8c98e-6803-4a51-ac2d-7fec8bbe77d3)

**Comment by francisdb** [comment]
This `Acrylics [Wall](11)`

<img width="573" alt="image" src="https://github.com/jsm174/vpx-standalone-scripts/assets/161305/ec74a31b-abf2-4cd4-8740-5425b2123966">

**Comment by francisdb** [comment]
Big bang bar has similar neon plastics

**Comment by jsm174** [comment]
I spent some time running this under Windows and Linux to get Renderdoc frame captures.

This is pretty much out of my domain, but it appears that in standalone, `GL_BLEND` is disabled where as in windows it is not.

![Screenshot_2023-06-17_at_5 55 25_PM](https://github.com/jsm174/vpx-standalone-scripts/assets/1197137/5b96adb1-3f78-4351-91d8-3b1dd38deee3)

I've let @vbousquet know, so hopefully he might have some more insights.

**Comment by francisdb** [comment]
Do we want to move this over to the vpinball repo then?

**Comment by jsm174** [comment]
If you want to, that would be great. I do have RDC captures if needed.

**Comment by francisdb** [comment]
I don't think I can move this ticket. Maybe you can?

**Comment by jsm174** [comment]
Turns out this was a long standing issue that I never noticed.

`GetMaterial` is one of the few VPX COM functions that have `[out]` parameters.  I did not have this coded correctly, so any tables that call GetMaterial would just get empty values, thus causing strange results.

This now renders fine in standalone:

![Screenshot 2023-07-20 at 5 08 09 PM](https://github.com/jsm174/vpx-standalone-scripts/assets/1197137/dea41d83-7769-4a7f-94db-353dcb799e35)
